### PR TITLE
Api features

### PR DESCRIPTION
**Pull request by mgax**
* show full path to a file
* click-browse through directories
* paginate the api document feed
* download original files

### CODE REVIEW

File: snoop/data/models.py [hunk 1]
```diff
@@ -66,6 +66,13 @@ class Blob(models.Model):
     def __str__(self):
         return self.pk
 
+    @property
+    def content_type(self):
+        if self.mime_type.startswith('text/'):
+            return f"{self.mime_type}; charset={self.mime_encoding}"
+
+        return self.mime_type
```
Comment (gabriel-v) [inline snoop/data/models.py]: why not always include the `charset`? or include it when it isn't `binary`?

File: snoop/data/digests.py [hunk 2]
```diff
@@ -65,15 +65,40 @@ def filetype(mime_type):
...
 def get_document_data(digest):
     with digest.result.open() as f:
         digest_data = json.loads(f.read().decode('utf8'))
 
     first_file = digest.blob.file_set.order_by('pk').first()
-    filename = path = first_file.name
 
     return {
         'id': digest.blob.pk,
+        'parent_id': parent_id(first_file),
```
Comment (gabriel-v) [inline snoop/data/digests.py]: if the "Parent" button only leads you to the first file, then how can you navigate to the other places where a document shows up?

File: snoop/data/urls.py [hunk 3]
```diff
@@ -1,8 +1,11 @@
 from django.urls import path
 from . import views
 
+
 urlpatterns = [
     path('<name>/json', views.collection),
     path('<name>/feed', views.feed),
+    path('<name>/_directory:<int:pk>/json', views.directory),
```
Comment (gabriel-v) [inline snoop/data/urls.py]: please don't put `:` in URLs because some browsers might behave badly

File: snoop/data/digests.py (hunk 2 above)
Comment (mgax) [reply snoop/data/digests.py]: You can't yet.

File: snoop/data/models.py (hunk 1 above)
Comment (mgax) [reply snoop/data/models.py]: I think it makes no sense unless the content is text. Do you have an example of a non-text mime type that has a charset? Or some documentation somewhere?

**Comment by mgax** [comment]
> This breaks the "Parent" button.

Au contraire, it makes the parent button work. I've been testing with the UI.

**Pull request merged**